pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -309,6 +310,7 @@ func testClusterDeployment() *hivev1.ClusterDeployment {
 		Spec: hivev1.ClusterDeploymentSpec{
 			ClusterUUID: testUUID,
 			Config: hivev1.InstallConfig{
+				ClusterID: testClusterID,
```

Comment:
Is a clusterID required?
If not, maybe you also want a test without it.
(we happened to use Hive with generateName and clusterID but what if someone used generateName and didn't specify cluaterID?)

Reply:
We have no validation framework yet, Thomas is working on it, but when we do ClusterID will be mandatory to get the object into etcd. I will make sure we have this on the Jira card as a required validation.